### PR TITLE
Python Lab: lazy load pyodide

### PR DESCRIPTION
**Pull request by molly-moen**
I noticed that Music Lab was loading pyodide. This is because is [LabViewsRenderer](https://github.com/code-dot-org/code-dot-org/blob/ff0a7237d29434fb3deeb22d470de4150c718807/apps/src/lab2/views/LabViewsRenderer.tsx#L46) we create all possible lab views, and Python Lab immediately loads pyodide. To fix this, I have pyodide only load when the run button is clicked. This does make the first run of python slower, but since this is a prototype that is fine. Long term we can probably find a different proactive way to load pyodide.

## Testing story
Tested locally. Now Music Lab does not load pyodide, but Python lab does on run.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/pythonlab/pyodideWebWorker.js [hunk 1]
```diff
@@ -12,11 +12,17 @@ async function loadPyodideAndPackages() {
   });
 }
 
-const pyodideReadyPromise = loadPyodideAndPackages();
+let pyodideReadyPromise = null;
+export async function initializePyodide() {
```
Comment (ebeastlake) [inline apps/src/pythonlab/pyodideWebWorker.js]: Was there a particular reason for exporting?

File: apps/src/pythonlab/pyodideWebWorker.js [hunk 2]
```diff
@@ -12,11 +12,17 @@ async function loadPyodideAndPackages() {
...
+  if (pyodideReadyPromise === null) {
+    pyodideReadyPromise = loadPyodideAndPackages();
+  }
+  await pyodideReadyPromise;
```
Comment (ebeastlake) [inline apps/src/pythonlab/pyodideWebWorker.js:20]: Remind my brain that's async/await rusty: If we've already awaited this promise once (on initial load) and it's resolved, this is functionally just a return statement, right?

**Review by ebeastlake — APPROVED** [review]
A couple of questions for my own knowledge, but nice catch and thanks for the fix!

File: apps/src/pythonlab/pyodideWebWorker.js (hunk 1 above)
Comment (molly-moen) [reply apps/src/pythonlab/pyodideWebWorker.js]: nope I'll fix, that was me working too quickly

File: apps/src/pythonlab/pyodideWebWorker.js (hunk 2 above)
Comment (molly-moen) [reply apps/src/pythonlab/pyodideWebWorker.js:20]: yep if it's already resolved it will return immediately

**Pull request merged**